### DISCUSSION TITLE
How can I achieve tree shaking in development mode?

### DISCUSSION THREAD

**Discussion by florinpavel22** [webpack/webpack#16038] [Q&A]
Hi everyone, I'm posting this with the hope that I'll get some help, as I've searched and researched a lot of websites without finding a working fix.

So, we have a monorepo with Lerna that includes different packages. Two of the most important ones are React components packages.
Let's call them `package-a` and `package-b`.
`package-b` is the newer one and depends on `package-a`, by importing some of its components and customizing them, according to the needs.

In short, this is the project structure:

```
packages
|__ **package-a**
|____ node_modules
|____ package.json
|____ webpack.config.js
|____ src
|_______ atoms
|__________ index.js
|__________ button.js
|__________ avatar.js
|_______ molecules
|__________ index.js
|__________ slider.js
|_______ index.js (exports all the components from the above folders (atoms & molecules))
|____ other-files

|__ **package-b**
|____ node_modules
|_______ package-a
|_______ other-node-modules
|____ package.json
|____ webpack.config.js
|____ src
|_______ atoms
|__________ index.js
|__________ different-button.js (here we can import the button from package-a, for example)
|_______ molecules
|__________ different-slider.js (here we can import the slider from package-a, for example)
|_______ index.js (exports all the components from atoms & molecules)
|____ other-files

|__ css-package
|__ svg-package
|__ other-packages
```

The problem is, I noticed that `package-b`'s bundle size is pretty big in development. If `package-a`'s size is around 10MB, `package-b`'s size is double. This created a bit of a headache when working with these packages, as the load time is quite big.

I noticed that in `package-b`'s bundle size, we get `package-a` bundle included, even if we import a single component from there. I tried different tips I've found while researching tree shaking, but nothing seems to bring the size down. Anyone can help me please?

**Comment by alexander-akait** [comment]
There are all options enabled in the production mode - https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L1128.
You can enable https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L1149 and https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L1150, and maybe https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L1147

**Comment by florinpavel22** [comment]
Hi, no change in build size with these configs enabled.

**Reply by alexander-akait** [reply]
There are other options https://github.com/webpack/webpack/blob/main/lib/config/defaults.js#L1129 you can try to enable/disable and look at size, note - some of things removed by terser, webpack just set them to `undefined`

**Comment by vankop** [comment]
@florinpavel22 please provide your dev webpack config
